public-ui/src/app/data-browser/views/survey-view/survey-view.component.ts

```diff
@@ -517,7 +517,19 @@ export class SurveyViewComponent implements OnInit, OnDestroy {
       });
   }
 
-  public showAnswerGraphs(a: any, q: any, level?: number) {
+  public showAnswerGraphs(a: any, q: any, level?: number, event?: MouseEvent) {
+    if (event && this.chartEl && this.chartEl.nativeElement.contains(event.target)) {
```

Comment:
This only works for one section per question since each element in the loop gets the same selector (sorry, didn't notice this when I first tested this workaround). If you expand a second answer under the same question, it will close when you try to click the graph tabs. It should work if we use `@ViewChildren` decorator instead.

On line 58:
`@ViewChildren('chartElement') chartEl: QueryList<ElementRef>;`

Then here, we need to check if any element in the list is clicked:
```suggestion
    if (event && this.chartEl && this.chartEl.some(el => el.nativeElement.contains(event.target))) {
```

This seems to work well, but after changing, I noticed if multiple answers are expanded, switching graph types in one answer will switch for all answers that are expanded under the same question:

https://user-images.githubusercontent.com/40036095/113972560-69f4b980-9800-11eb-8912-66dd3cc29643.mov

Reply:
Thanks for this fix. 
It also looks like all the charts have same analysis when multiple charts are opened. So seems like the problems with shared state.